ruma-api/src/lib.rs

```diff
@@ -289,6 +289,17 @@ pub trait OutgoingNonAuthRequest: OutgoingRequest {}
 /// Marker trait for requests that don't require authentication. (for the server side)
 pub trait IncomingNonAuthRequest: IncomingRequest {}
 
+/// Authentication scheme used by the endpoint.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum AuthenticationStrategy {
+    /// No authentication is performed.
+    None,
+    /// Authentication is performed by including an access token in the request headers.
+    AccessToken,
+    /// Authentication is performed by including an X-Matrix signature in the request headers, as defined in the server-server API.
+    ServerServerHeader,
```

Comment:
Since the server-server API is mostly referred to as the federation API and that's also how we call its api crate, maybe we should rename to FederationHeader. CC @iinuwa

Reply:
Haha, this is the first time I've read this part of the spec. :laughing: 

Since the header is still the `Authorization` header, I think we should either use the scheme name or a description of its content for the name: e.g., either `XMatrix` or `ServerSignatures`.

Reply:
There's only one signature per request though, so the extra `s` at the end of `ServerSignatures` seems pointless (am I missing something? just a typo?). I think `ServerSignature` would be good :+1:

Reply:
The example Python code has a `for` loop that made me think there could be multiple signatures (i.e. signed by the same origin server with multiple keys.)
```python
authorization_headers = []

for key, sig in signed_json["signatures"][origin_name].items():
    authorization_headers.append(bytes(
        "X-Matrix origin=%s,key=\"%s\",sig=\"%s\"" % (
            origin_name, key, sig,
        )
    ))

return ("Authorization", authorization_headers)
```

Reply:
Makes sense. So `ServerSignatures` as the name and

> including X-Matrix signatures

in the doc comment.

Reply:
Oh, please wrap any (doc) comments at 100 chars by the way @williewillus